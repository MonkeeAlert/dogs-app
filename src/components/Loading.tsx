import {
  ActivityIndicator,
  ActivityIndicatorProps,
  StyleProp,
  StyleSheet,
  View,
  ViewStyle,
} from 'react-native';
import React from 'react';

interface Props {
  size?: ActivityIndicatorProps['size'];
  style?: StyleProp<ViewStyle>;
}

const Loading = ({size = 'large', style = {}}: Props) => {
  return (
    <View testID={'Loading_View'} style={[styles.container, style]}>
      <ActivityIndicator testID={'Loading_Indicator'} size={size} />
    </View>
  );
};

export default Loading;

const styles = StyleSheet.create({
  container: {flex: 1, alignItems: 'center', justifyContent: 'center'},
});
