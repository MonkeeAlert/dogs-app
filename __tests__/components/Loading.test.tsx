import React from 'react';
import renderer from 'react-test-renderer';
import Loading from '../../src/components/Loading';
import {shallow} from 'enzyme';

describe('Loading', () => {
  it('snapshot', () => {
    const tree = renderer.create(<Loading />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with default props', () => {
    const wrapper = shallow(<Loading />);

    const view = wrapper.find({testID: 'Loading_View'}).first();
    expect(view.exists()).toBe(true);

    const viewStyles = view.props().style;
    expect(Object.keys(viewStyles[0]).length).toBeGreaterThan(0);
    expect(Object.keys(viewStyles[1]).length).toBe(0);

    const indicator = wrapper.find({testID: 'Loading_Indicator'}).first();
    expect(indicator.exists()).toBe(true);

    const indicatorSize = indicator.props().size;
    expect(indicatorSize).toBe('large');
  });

  it('renders with style prop', () => {
    const wrapper = shallow(<Loading style={{backgroundColor: 'red'}} />);

    const view = wrapper.find({testID: 'Loading_View'}).first();
    expect(view.exists()).toBe(true);

    const viewStyles = view.props().style;
    expect(Object.keys(viewStyles[0]).length).toBeGreaterThan(0);
    expect(Object.keys(viewStyles[1]).length).toBeGreaterThan(0);
  });

  it('renders with size prop (string)', () => {
    const wrapper = shallow(<Loading size={'small'} />);
    const expected = ['small', 'large'];

    const indicator = wrapper.find({testID: 'Loading_Indicator'}).first();
    expect(indicator.exists()).toBe(true);

    const indicatorSize = indicator.props().size;
    expect(expected.includes(indicatorSize)).toBe(true);
  });

  it('renders with size prop (number)', () => {
    const wrapper = shallow(<Loading size={20} />);

    const indicator = wrapper.find({testID: 'Loading_Indicator'}).first();
    expect(indicator.exists()).toBe(true);

    const indicatorSize = indicator.props().size;
    expect(indicatorSize).toBe(20);
  });
});
