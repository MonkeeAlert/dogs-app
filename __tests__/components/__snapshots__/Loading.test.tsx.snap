// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {},
    ]
  }
  testID="Loading_View"
>
  <ActivityIndicator
    size="large"
    testID="Loading_Indicator"
  />
</View>
`;
